server/api.go

```diff
@@ -495,6 +505,116 @@ func (p *Plugin) getYourPrs(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func (p *Plugin) getPrsDetails(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	var prList []*PRDetails
+	json.NewDecoder(r.Body).Decode(&prList)
+
+	var prDetails []*PRDetails
+
+	ch := make(chan *PRDetails)
+	for _, pr := range prList {
+		go fetchPRDetails(ctx, githubClient, pr.URL, pr.Number, ch)
+	}
+
+	for range prList {
+		pr := <-ch
+		prDetails = append(prDetails, pr)
+	}
+
+	resp, _ := json.Marshal(prDetails)
+	w.Write(resp)
+}
+
+func fetchPRDetails(ctx context.Context, client *github.Client, prURL string, prNumber int, ch chan *PRDetails) {
+	status := ""
+	requestedReviewers := []*string{}
+	var reviewsList []*github.PullRequestReview = nil
+
+	repoOwner, repoName := getRepoOwnerAndNameFromURL(prURL)
+
+	chReviews := make(chan []*github.PullRequestReview)
+
+	go fetchReviews(ctx, client, repoOwner, repoName, prNumber, chReviews)
+
+	prInfo, _, err := client.PullRequests.Get(ctx, repoOwner, repoName, prNumber)
+
+	if err != nil {
```

Comment:
I think we should delay the call to `fetchReviews` until after we know this `err` is nil.

Reply:
The reason behind this is that fetchReviews will take some time, as getting the pull requests. Doing it in parallel would speed things up. Nevertheless, I see now that there might be problems of goroutines stopped on a channel in case this err occurs.

Would you like me to work with this order and solve this problem with the goroutines and channels, or do you prefer changing the order?